lighthouse-core/audits/seo/hreflang.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+const LinkHeader = require('http-link-header');
+const axeCore = require('axe-core');
+const LINK_HEADER = 'link';
+const NO_LANGUAGE = 'x-default';
+
+/**
+ * @param {String} hreflang 
+ * @returns {Boolean}
+ */
+function isValidHreflang(hreflang) {
+  if (hreflang.toLowerCase() === NO_LANGUAGE) {
+    return true;
+  }
+
+  // hreflang can consist of language-script-region, we are validating only language
+  const [lang] = hreflang.split('-');
+  return axeCore.utils.validLangs().includes(lang.toLowerCase());
+}
+
+/**
+ * @param {String} headerValue 
+ * @returns {Boolean}
+ */
+function headerHasValidHreflangs(headerValue) {
+  const linkHeader = LinkHeader.parse(headerValue);
+
+  return linkHeader.get('rel', 'alternate')
```

Comment:
always returns an array? strange api 🤔

Reply:
Single `Link` header can have multiple `rel=alternate` links (e.g. `<http://es.example.com/>; rel="alternate"; hreflang="es",<http://fr.example.com/>; rel="alternate"; Hreflang="fr-be"`), so IMO it makes sense to always return an array.